### ISSUE TITLE
Getting errors through take_while! or similar loop

### ISSUE THREAD

**Issue by kaj** [Geal/nom#463]
I have something that looks like this:

    many_till!(complex_subparser,
               simple_end_marker)

What I was hoping to get is something like the following pseudocode:

    let mut result = vec![];
    while(simple_end_marker is not matching) {
        result.append(try!(complex_subparser))
    }
    result

That is, as long as the simple_end_marker does not match, I expect the complex_subparser to match. If it doesn't, I want the error from it.  Currently, if both the simple_end_marker and the complex_subparser fails, all I get is a `ManyTill` error.

I'm using the `verbose-errors` feature.

My actual code, for context: https://github.com/kaj/ructe/blob/feature/improve-error-handling/src/template.rs#L63

**Comment by juggle-tux** [comment]
maybe this could work
`separated_nonempty_list!(simple_end_marker, complex_subparser)`

**Comment by kaj** [comment]
@juggle-tux , no, that would require the `simple_end_marker` between each `complex_subparser` instead of a single `simple_end_marker` after them all.  Hm.  But maybe I can create a non-consuming negated version of the simple_end_marker?  Since my end markers are generally just eof or the '}' character, I guess that should be easy enogh.

Thanks for the suggestion, @juggle-tux !  :-)
